### PR TITLE
Add deprecation warning about connectors with arbitrary names

### PR DESCRIPTION
**Pull request by spacecowboy**
This time in java, it still needs a little work because this is the actual output:

```
$ bin/neo4j console
Starting Neo4j.
WARNING: Max 1024 open files allowed, minimum of 40000 recommended. See the Neo4j manual.
2017-02-01 16:02:24.962+0000 WARN  Warning: connectors with names other than [http,https,bolt] are
deprecated and support for them will be removed in a future
version of Neo4j. Offending lines in neo4j.conf:

>  dbms.connector.jonas.enabled
>  dbms.connector.jonas.type

2017-02-01 16:02:25.004+0000 INFO  No SSL certificate found, generating a self-signed certificate..
2017-02-01 16:02:25.450+0000 INFO  Starting...
2017-02-01 16:02:25.465+0000 WARN  Warning: connectors with names other than [http,https,bolt] are
deprecated and support for them will be removed in a future
version of Neo4j. Offending lines in neo4j.conf:

>  dbms.connector.jonas.enabled
>  dbms.connector.jonas.type

2017-02-01 16:02:25.661+0000 WARN  Warning: connectors with names other than [http,https,bolt] are
deprecated and support for them will be removed in a future
version of Neo4j. Offending lines in neo4j.conf:

>  dbms.connector.jonas.enabled
>  dbms.connector.jonas.type

2017-02-01 16:02:26.047+0000 INFO  Bolt enabled on localhost:7687.
2017-02-01 16:02:26.055+0000 WARN  Warning: connectors with names other than [http,https,bolt] are
deprecated and support for them will be removed in a future
version of Neo4j. Offending lines in neo4j.conf:

>  dbms.connector.jonas.enabled
>  dbms.connector.jonas.type

2017-02-01 16:02:26.059+0000 INFO  Initiating metrics...
2017-02-01 16:02:26.128+0000 WARN  Warning: connectors with names other than [http,https,bolt] are
deprecated and support for them will be removed in a future
version of Neo4j. Offending lines in neo4j.conf:

>  dbms.connector.jonas.enabled
>  dbms.connector.jonas.type

2017-02-01 16:02:28.222+0000 INFO  Started.
2017-02-01 16:02:28.347+0000 INFO  Mounted REST API at: /db/manage
2017-02-01 16:02:28.925+0000 INFO  Remote interface available at http://localhost:7474/
```

### CODE REVIEW

File: community/graphdb-api/src/main/java/org/neo4j/graphdb/config/SettingValidator.java [hunk 1]
```diff
@@ -20,16 +20,19 @@
...
      * Validate one or several setting values, throwing on invalid values.
      *
      * @param settings available to be validated
+     * @param warningConsumer a consumer for configuration warnings
      * @return the set of settings considered valid by this validator
      * @throws InvalidSettingException if invalid value detected
      */
-    Map<String,String> validate( Map<String,String> settings ) throws InvalidSettingException;
+    Map<String,String> validate( Map<String,String> settings, Consumer<String> warningConsumer )
+            throws InvalidSettingException;
```
Comment (benbc) [inline community/graphdb-api/src/main/java/org/neo4j/graphdb/config/SettingValidator.java:36]: Is this method new in 3.2?

**Comment by benbc** [comment]
"Still needs a little work" because of the duplicated warnings?

File: community/graphdb-api/src/main/java/org/neo4j/graphdb/config/SettingValidator.java (hunk 1 above)
Comment (spacecowboy) [reply community/graphdb-api/src/main/java/org/neo4j/graphdb/config/SettingValidator.java:36]: It was added as part of my huge config work

**Comment by spacecowboy** [comment]
Fixed the duplication in latest commit:

```
$ bin/neo4j console
Starting Neo4j.
WARNING: Max 1024 open files allowed, minimum of 40000 recommended. See the Neo4j manual.
2017-02-01 16:17:25.106+0000 WARN  Warning: connectors with names other than [http,https,bolt] are
deprecated and support for them will be removed in a future
version of Neo4j. Offending lines in neo4j.conf:

>  dbms.connector.jonas.enabled
>  dbms.connector.jonas.type

2017-02-01 16:17:25.146+0000 INFO  No SSL certificate found, generating a self-signed certificate..
2017-02-01 16:17:25.521+0000 INFO  Starting...
2017-02-01 16:17:26.100+0000 INFO  Bolt enabled on localhost:7687.
2017-02-01 16:17:26.116+0000 INFO  Initiating metrics...
2017-02-01 16:17:28.255+0000 INFO  Started.
2017-02-01 16:17:28.382+0000 INFO  Mounted REST API at: /db/manage
2017-02-01 16:17:28.955+0000 INFO  Remote interface available at http://localhost:7474/
```

**Comment by benbc** [comment]
Looks good to me.

**Pull request merged**